test/eventmesh.Etcd3EventMeshServer.spec.js

```diff
@@ -0,0 +1,210 @@
+          });
+      });
+    });
+
+    describe('#annotateResource', () => {
+      it('should annotate the resourse with option and state keys', () => {
+        return eventmesh.server.annotateResource('fakeResourceType', 'fakeResourceId', 'fakeAnnotationName', 'fakeOperationType', 'fakeOperationId', 'fakeVal')
+          .then(() => {
+            expect(putstub.args[0][0]).to.equal('deployments/fakeResourceType/fakeResourceId/fakeAnnotationName/fakeOperationType/fakeOperationId/options');
+            expect(putstub.args[1][0]).to.equal('deployments/fakeResourceType/fakeResourceId/fakeAnnotationName/fakeOperationType/fakeOperationId/state');
+            expect(valueStub.args[0][0]).to.equal('fakeVal');
+            expect(valueStub.args[1][0]).to.equal(CONST.RESOURCE_STATE.IN_QUEUE);
+          });
+      });
+    });
+
+    describe('#updateAnnotationState', () => {
+      it('should update the annotation state key', () => {
+        return eventmesh.server.updateAnnotationState('fakeResourceType', 'fakeResourceId', 'fakeAnnotationName', 'fakeOperationType', 'fakeOperationId', 'fakeStateValue')
+          .then(() => {
+            expect(putstub.args[0][0]).to.equal('deployments/fakeResourceType/fakeResourceId/fakeAnnotationName/fakeOperationType/fakeOperationId/state');
+            expect(valueStub.args[0][0]).to.equal('fakeStateValue');
+          });
+      });
+    });
+
+    describe('#updateAnnotationKey', () => {
+      it('should update the annotation key specified', () => {
+        return eventmesh.server.updateAnnotationKey('fakeResourceType', 'fakeResourceId', 'fakeAnnotationName', 'fakeOperationType', 'fakeOperationId', 'fakeKey', 'fakeValue')
+          .then(() => {
+            expect(putstub.args[0][0]).to.equal('deployments/fakeResourceType/fakeResourceId/fakeAnnotationName/fakeOperationType/fakeOperationId/fakeKey');
+            expect(valueStub.args[0][0]).to.equal('fakeValue');
+          });
+      });
+    });
+
+    describe('#getAnnotationKey', () => {
+      it('should get the annotation key', () => {
+        return eventmesh.server.getAnnotationKey('fakeResourceType', 'fakeResourceId', 'fakeAnnotationName', 'fakeOperationType', 'fakeOperationId', 'fakeKey')
+          .then(() => {
+            expect(getstub.args[0][0]).to.equal('deployments/fakeResourceType/fakeResourceId/fakeAnnotationName/fakeOperationType/fakeOperationId/fakeKey');
+          });
+      });
+    });
+
+    describe('#getAnnotationState', () => {
+      it('should get the annotation state', () => {
+        return eventmesh.server.getAnnotationState('fakeResourceType', 'fakeResourceId', 'fakeAnnotationName', 'fakeOperationType', 'fakeOperationId')
+          .then(() => {
+            expect(getstub.args[0][0]).to.equal('deployments/fakeResourceType/fakeResourceId/fakeAnnotationName/fakeOperationType/fakeOperationId/state');
+          });
+      });
+    });
+
+  });
+});
```

Comment:
Should we also not check for the response to be in specific format (in every function)? E.g. in 'registerService' function, response should be from second put, but if I add one more 'then' and return etcd.get(), the response will change but unit test will still pass.

Reply:
done